test/serializer/test_normalize.py

```diff
@@ -0,0 +1,162 @@
+
+    profile = Profile('12345678', Name('Foo Bar', 'Bar, Foo'), '0000-0002-1825-0097')
+    profile.add_email_address('1@example.com')
+    profile.add_email_address('2@example.com')
+    profile.add_email_address(primary_address, primary=True)
+
+    normalized_profile = normalize(profile)
+
+    assert len(normalized_profile['emailAddresses']) == 3
+    assert normalized_profile['emailAddresses'][0] == primary_address
+
+
+def test_it_normalizes_profile_with_an_affiliation():
+    address = Address(countries.get('gb'), 'City', 'Region')
+    affiliation = Affiliation('1', address=address, organisation='Org',
+                              department='Dep', starts=datetime.now())
+    profile = Profile('12345678', Name('Foo Bar', 'Bar, Foo'))
+
+    profile.add_affiliation(affiliation)
+
+    assert normalize(profile) == {
+        'id': '12345678',
+        'name': {
+            'preferred': 'Foo Bar',
+            'index': 'Bar, Foo',
+        },
+        'emailAddresses': [],
+        'affiliations': [
+            {
+                "name": [
+                    "Dep",
+                    "Org"
+                ],
+                "address": {
+                    "formatted": [
+                        "City",
+                        "Region",
+                        "GB"
+                    ],
+                    "components": {
+                        "locality": [
+                            "City"
+                        ],
+                        "area": [
+                            "Region"
+                        ],
+                        "country": "United Kingdom of Great Britain and Northern Ireland"
+                    }
+                }
+            }
+        ]
+    }
+
+
+def test_it_normalizes_profile_with_affiliations():
+    address = Address(countries.get('gb'), 'City', 'Region')
+    address2 = Address(countries.get('gb'), 'City2', 'Region2')
+    affiliation = Affiliation('1', address=address, organisation='Org',
+                              department='Dep', starts=datetime.now())
+    affiliation2 = Affiliation('2', address=address2, organisation='Org2',
+                               department='Dep', starts=datetime.now())
+
+    profile = Profile('12345678', Name('Foo Bar', 'Bar, Foo'))
+
+    profile.add_affiliation(affiliation)
+    profile.add_affiliation(affiliation2)
+
+    assert normalize(profile) == {
+        'id': '12345678',
+        'name': {
+            'preferred': 'Foo Bar',
+            'index': 'Bar, Foo',
+        },
+        'emailAddresses': [],
+        'affiliations': [
+            {
+                "name": ["Dep", "Org2"],
+                "address": {
+                    "formatted": ["City2", "Region2", "GB"],
+                    "components": {
+                        "locality": ["City2"],
+                        "area": ["Region2"],
+                        "country": "United Kingdom of Great Britain and Northern Ireland"
```

Comment:
Wonder if we can get a short name out of the library (ie `United Kingdom`)...

Reply:
The only non abbreviated or numeric fields are `name` (currently being used) or `apolitical_name` which seems to be Identical in the library:
```
Country(u"United Kingdom of Great Britain and Northern Ireland",
            "GB", "GBR", "826",
            u"United Kingdom of Great Britain and Northern Ireland"),
```
Just seems that the entry for the UK is rather wordy.

Reply:
Hmm, we might need to have our own set of names.